src/fides/api/alembic/migrations/versions/c9c72b3d550b_data_migration_converting_existing_.py

```diff
@@ -0,0 +1,214 @@
+        sa.ForeignKeyConstraint(
+            ["privacy_declaration_id"], ["privacydeclaration.id"], ondelete="SET NULL"
+        ),
+        sa.ForeignKeyConstraint(["system_id"], ["ctl_systems.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint(
+            "name", "privacy_declaration_id", name="_cookie_name_privacy_declaration_uc"
+        ),
+    )
+    op.create_index(op.f("ix_cookies_id"), "cookies", ["id"], unique=False)
+    op.create_index(op.f("ix_cookies_name"), "cookies", ["name"], unique=False)
+    op.create_index(
+        op.f("ix_cookies_privacy_declaration_id"),
+        "cookies",
+        ["privacy_declaration_id"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_cookies_system_id"), "cookies", ["system_id"], unique=False
+    )
+    # ### end Alembic commands ###
+
+    connection = op.get_bind()
+    result = connection.execute(
+        sa.text(
+            "SELECT id, name, domain, system_id, data_uses FROM asset WHERE asset_type = 'Cookie'"
+        )
+    )
+
+    logger.debug("Migrating existing assets to cookies")
+    for row in result:
+        # Find the privacy declaration ID matching the system ID and data use
+        privacy_declaration_ids = []
+        for data_use in row.data_uses:
+            pud = connection.execute(
+                sa.text(
+                    "SELECT id FROM privacydeclaration WHERE system_id = :system_id AND data_use = :data_use"
+                ),
+                system_id=row.system_id,
+                data_use=data_use,
+            ).fetchone()
+            if pud:
+                privacy_declaration_ids.append(pud.id)
+
+        if len(privacy_declaration_ids) == 0:
+            # If no privacy declaration match we attach to system_id
+            connection.execute(
+                sa.text(
+                    "INSERT INTO cookies (id, created_at, updated_at, name, domain, system_id) "
+                    "VALUES (:id, NOW(), NOW(), :name, :domain, :system_id)"
+                    "ON CONFLICT DO NOTHING"
+                ),
+                id=row.id,
+                name=row.name,
+                domain=row.domain,
+                system_id=row.system_id,
+            )
+        for privacy_declaration_id in privacy_declaration_ids:
+            # Create a cookie for each privacy declaration ID
+            # generate a new ID
+            cookie_id = str(uuid.uuid4())
+            connection.execute(
+                sa.text(
+                    "INSERT INTO cookies (id, created_at, updated_at, name, domain, privacy_declaration_id) "
+                    "VALUES (:id, NOW(), NOW(), :name, :domain, :privacy_declaration_id)"
+                    "ON CONFLICT DO NOTHING"
+                ),
+                id=cookie_id,
+                name=row.name,
+                domain=row.domain,
+                privacy_declaration_id=privacy_declaration_id,
```

Comment:
again, just want to make sure we do a bulk add here rather than one query per asset. (i know i'm a broken record, but erring on the side of thoroughness even at the expense of being repetitive 😅 )

Reply:
Addressed here: https://github.com/ethyca/fides/pull/5776/commits/54e4eac935d25653649ec9f2424b76047206c7bb